packages/interactor/src/selector.ts

```diff
@@ -0,0 +1,19 @@
+import { when } from '~/when';
+import { throwIfEmpty } from '~/util';
+
+type Query<Container, Result> = (
+  locator: string,
+  container: Container
+) => Iterable<Result> | Promise<Iterable<Result>>;
+
+export type Selector<Container, Result> = (locator: string, container: Container) => Promise<Result[]>;
+
+export function selector<Container, Result>(query: Query<Container, Result>): Selector<Container, Result> {
+  return (locator, container) =>
+    when(async () =>
+      throwIfEmpty(
+        Array.from(await query(locator, container)),
+        `Did not find any matches with locator "${locator}"`
+      )
```

Comment:
This is fine as a first implementation, but based on my experience from Capybara, I'm thinking that this will probably be too lax, eventually. Capybara gives you the option of specifying the conditions [when a selector is allowed to proceed](https://github.com/teamcapybara/capybara#strategy). And by default it is not allowed to have more than one matching element. This is kind of important for making tests more solid, since sometimes you might have several matching elements on the page, and it should be deterministic which of those you click on.